src/main/scala/com/microsoft/hyperspace/actions/RefreshActionBase.scala

```diff
@@ -128,7 +126,15 @@ private[actions] abstract class RefreshActionBase(
             _) =>
           location
             .allFiles()
-            .map(f => FileInfo(f.getPath.toString, f.getLen, f.getModificationTime))
+            .map { f => // for each file, if it already has a file id,
+                        // add that id to its corresponding FileInfo.
+              val filePath = f.getPath.toString
+              previousIndexLogEntry.fileIdsMap.get(filePath) match {
+                case Some(id) =>
+                  FileInfo(filePath, f.getLen, f.getModificationTime, id)
+                case _ =>
+                  FileInfo(filePath, f.getLen, f.getModificationTime)
```

Comment:
ditto: can we assign a new id here? so that all FileInfo has an id.

Reply:
Plz check my answer to the other comment.